### PR TITLE
Custom View DTOs

### LINKED ISSUE #432: View Entity DTOs
Create DTOs to spec out the interface and data shapes expected and returned by the API endpoints.

- Create `CreateViewDTO`. This should contain all the properties of [`View`](https://seas-computing.github.io/course-planner/classes/view.html) except the `eppn` property.
This DTO represents the shape of data expected by the creation endpoint (`POST`).
- Create `UpdateViewDTO`. This should be of type `Partial<View>` minus the `eppn` property.
This DTO represents the shape of data expected by the update endpoint (`PUT`).
- Create `ViewResponseDTO`. This should more or less mirror CreateViewDTO and should be used to indicate the data shape returned by the API to represent View entities
This DTO represents the shape of data expected by the report endpoint (`GET`).

### PR DESCRIPTION
**Pull request by rmainseas**
This PR delivers #432 and specs out the data shapes for the controller CRUD endpoints for custom view functionality on the course admin page
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality
      to change)

## Checklist:

- [x] I have run `eslint` on the code
- [x] I have added JSDoc for all of my code (where applicable)
- [ ] I have added tests to cover my changes.

## Priority:
- [x] Normal 
- [ ] High 

## Related Issues:

Fixes #432

### CODE REVIEW

File: src/common/dto/view/CreateView.dto.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  @IsEnum(COURSE_TABLE_COLUMN, { each: true })
+  public columns: COURSE_TABLE_COLUMN[] = [];
+
+  @ApiProperty({
+    example: 'No semester data',
+    description: 'A descriptive name for this custom view',
+  })
+  @IsString()
+  @MinLength(1)
+  public name: string;
```
Comment (natalynyu) [inline src/common/dto/view/CreateView.dto.ts:29]: We have a decorator that trims the string (`@Transform(trimString)`) that's being used in the `CreateCourse` dto that could be used here along with for the other `name` properties in order to sanitize the text input from the user.

File: src/common/dto/view/CreateView.dto.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * Defines the data shape for creating a new [[View]] entry in the database.
+ */
+export class CreateViewDTO {
+  @ApiProperty({
+    description: 'Array of column names for this view',
+    example: COURSE_TABLE_COLUMN.AREA,
+    enum: COURSE_TABLE_COLUMN,
+  })
+  @IsEnum(COURSE_TABLE_COLUMN, { each: true })
+  public columns: COURSE_TABLE_COLUMN[] = [];
```
Comment (natalynyu) [inline src/common/dto/view/CreateView.dto.ts:21]: I was going to suggest making sure the array is not empty, but I suppose it should be okay for someone to create a view with no columns at first if they wanted. What do you think?

**Review by natalynyu — COMMENTED** [review]
Looks great! Just a few small comments

File: src/common/dto/view/CreateView.dto.ts (hunk 2 above)
Comment (rmainseas) [reply src/common/dto/view/CreateView.dto.ts:21]: Yeah, as much as it doesn't make sense logically - I think we might want to allow this.

File: src/common/dto/view/CreateView.dto.ts (hunk 1 above)
Comment (rmainseas) [reply src/common/dto/view/CreateView.dto.ts:32]: Fixed that in b3f67944610c487c6f84d6e08782e15138818a54

**Review by natalynyu — APPROVED** [review]
Looks good to me! Approved.

**Pull request merged**